rasa/engine/graph.py

```diff
@@ -104,6 +128,8 @@ def __init__(
         fn_name: Text,
         inputs: Dict[Text, Text],
         eager: bool,
+        model_storage: ModelStorage,
+        resource_name: Optional[Text],
```

Comment:
You mean wrapping it in `graph_node.from_schema_node`? Can do 👍🏻

Reply:
Or even having it in the `SchemaNode` too? That was what I initially expected.

Reply:
actually - considering that we pass in the `node_name` as string as well I think it makes more sense to keep all `Resource(...)` wrapping in here instead of spreading it out (related to your next comment)

Reply:
> Or even having it in the SchemaNode too? That was what I initially expected.

What's the advantage of leaking the `Resource` type even further outside?

Reply:
but we are passing `node_name` into a `GraphNode` in the same way a `Resource` has a `name`. We also pass `Resource`s around the graph - doesn't it make sense to use something typed?

Reply:
done